lib/platform.py

```diff
@@ -53,28 +55,30 @@
-  HEVC_DECODE_10BIT_PLATFORMS   = [                                          "APL", "KBL",        "CFL", "WHL", "ICL"]
-  HEVC_ENCODE_10BIT_PLATFORMS   = [                                                 "KBL",        "CFL", "WHL", "ICL"]
-  HEVC_ENCODE_10BIT_LP_PLATFORMS= [                                                                             "ICL"]
-  VP8_DECODE_PLATFORMS          = [                     "BDW",        "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
-  VP8_ENCODE_PLATFORMS          = [                                                                             "ICL"]
-  VP9_DECODE_8BIT_PLATFORMS     = [                                          "APL", "KBL",        "CFL", "WHL", "ICL"]
-  VP9_ENCODE_8BIT_PLATFORMS     = [                                                                             "ICL"]
-  VP9_DECODE_10BIT_PLATFORMS    = [                                                 "KBL",        "CFL", "WHL", "ICL"]
-  VP9_ENCODE_10BIT_PLATFORMS    = [                                                                             "ICL"]
-  VPP_PLATFORMS                 = [                     "BDW",        "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  JPEG_DECODE_PLATFORMS           = [                   "BDW",        "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  JPEG_ENCODE_PLATFORMS           = [                                 "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  MPEG2_DECODE_PLATFORMS          = [                   "BDW",        "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  MPEG2_ENCODE_PLATFORMS          = [                   "BDW",        "SKL",        "KBL",        "CFL", "WHL", "ICL"]
+  VC1_DECODE_PLATFORMS            = [                   "BDW",        "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  AVC_DECODE_PLATFORMS            = [                   "BDW",        "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  AVC_ENCODE_PLATFORMS            = [                   "BDW",        "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  AVC_ENCODE_LP_PLATFORMS         = [                                 "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  HEVC_DECODE_8BIT_PLATFORMS      = [                                 "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  HEVC_DECODE_8BIT_8K_PLATFORMS   = [                                 "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  HEVC_ENCODE_8BIT_PLATFORMS      = [                                 "SKL", "APL", "KBL",        "CFL", "WHL", "ICL"]
+  HEVC_ENCODE_8BIT_LP_PLATFORMS   = [                                                                           "ICL"]
+  HEVC_DECODE_10BIT_PLATFORMS     = [                                        "APL", "KBL",        "CFL", "WHL", "ICL"]
+  HEVC_DECODE_10BIT_422_PLATFORMS = [                                                                           "ICL"]
+  HEVC_DECODE_10BIT_444_PLATFORMS = [                                                                           "ICL"]
```

Comment:
Unfortunately, the original idea was to use these only for `codec->platform` mappings.  Now we're starting to see `resolution->platform` and `pix format->platform` mappings creep into this design.  This is making it worse to maintain (i.e. it forces us to add more specialized tags, test functions and user config keys).  Maybe we should start thinking about a better way to handle these latter cases at test runtime.  We don't have to do it now, but let's think about it later.  I have a few ideas.

Your current solution is ok for now.

Reply:
how about this
DECODE_10BIT_422_PLATFORMS = [ ... ]
@platform_tags(set(HEVC_DECODE_10BIT_PLATFORMS) & set(DECODE_10BIT_422_PLATFORMS))

Reply:
yes, that is better.